src/fortran/dataset.F90

```diff
@@ -478,6 +480,33 @@ function get_tensor_type(self, name, ttype) result(code)
   code = get_tensor_type_c(self%dataset_ptr, c_name, name_length, ttype)
 end function get_tensor_type
 
+
+!> Retrieve the dimensions for a tensor into a supplied buffer, or receive the
+!! number of dimensions if the supplied buffer is too small
+function get_tensor_dims(self, name, dims, dims_length) result(code)
+  class(dataset_type),     intent(in)    :: self  !< The dataset
+  character(len=*),        intent(in)    :: name  !< The name of the tensor
+  integer(kind=c_size_t), dimension(:), target, intent(inout) :: dims !< Receives the tensor dimensions
+  integer(kind=c_size_t),  intent(inout) :: dims_length !< Receives the number of tensor dimensions
+  integer(kind=enum_kind)                :: code  !< Result of the operation
+
+  ! local variables
+  character(kind=c_char, len=len_trim(name)) :: c_name
+  integer(kind=c_size_t) :: name_length
+  type(c_ptr) :: dims_ptr
+
+  c_name = trim(name)
+  name_length = len_trim(c_name)
+
+  if (dims_length .gt. size(dims)) then
+    error stop 'undersized buffer in call to get_tensor_dims'
```

Comment:
Add the minimum size of `dims` that the user would need to specify to the error message

Reply:
We don't know it yet. This is the case where they pass us a buffer for the dims, say of size 3 and then tell us it's bigger than that, say size 10. If I ignore that, we risk having memory corruption and a segfault, so I want to warn them straightaway. I could just silently truncate down num_dims before proceeding but I'd rather make the caller aware of the error they've made.

Reply:
Made the error message clearer